Enable https for all services and create dedicated ssl pillar profile

mjura opened the pull request:
This change is enabling https for etcd-proxy and flanneld services and it is cleaning up pillar configuration for ssl.

This is basically PR https://github.com/kubic-project/salt/pull/86

flavio commented:
We have a bugzilla entry for this issue: bsc#1043595

Please update the commits to include the bsc into the subject

kiall commented:
Also see #108 - where a few other bits were removed.

PI-Victor commented:
this seems to be tied to a P1, is there an update for this?

mjura commented:
I've rebased changes on top of master, fixed some minor issues and tested it on my dev environment using terraform project
```
export FLAVOUR=caasp
export LATEST_IMAGE=true

contrib/libvirt/k8s-libvirt.sh apply
```

mjura commented:
Works fine, let's wait for e2e tests

ereslibre commented on salt/flannel/flanneld.sysconfig.jinja:
```diff
@@ -1,12 +1,21 @@
+{% set etcd_ca = "-etcd-cafile " + pillar['ssl']['ca_file'] -%}
+{% set etcd_key = "-etcd-keyfile " + pillar['ssl']['key_file'] -%}
+{% set etcd_crt = "-etcd-certfile " + pillar['ssl']['crt_file'] -%}
+{% set flannel_opt = flannel_opt + " " + etcd_ca + " " + etcd_key + " " + etcd_crt -%}
```
Could you please indent this set of lines, for better readability.

mjura replied:
Ok, good point, done

barclaac commented on salt/etcd-proxy/etcd-proxy.conf.jinja:
```diff
@@ -4,19 +4,19 @@
+ETCD_PEER_CERT_FILE={{ pillar['ssl']['crt_file'] }}
+ETCD_PEER_KEY_FILE={{ pillar['ssl']['key_file'] }}
+ETCD_PEER_TRUSTED_CA_FILE={{ pillar['ssl']['ca_file'] }}
 # ETCD_PEER_CLIENT_CERT_AUTH=on
```
Shouldn't CLIENT_CERT_AUTH = on. Looks like this line needs to be uncommented.

barclaac replied:
And I just checked in the docs. Both ETCD_PEER_CLIENT_CERT_AUTH and ETCD_CLIENT_CERT_AUTH default to false so these will need to be set explicitly.

mjura replied:
Thank you for your comment. Options `ETCD_PEER_CLIENT_CERT_AUTH` and `ETCD_CLIENT_CERT_AUTH` are enabling authentication based on certificates, which we don't have it. In our configuration we have to use only encrypted connection.

kiall replied:
Looking at bsc#1043595, sure, it's not explicitly called out. However, it absolutely has to be enabled as transport encryption without authentication doesn't provide much value - e.g. why sniff traffic if you can just connect in yourself and direct ask for the data you want?

mjura replied:
@kiall: This change is only about enabling secure connection to ETCD cluster, for authentication we have another bug bsc#1044590

kiall replied:
I disagree, bsc#1044590 is now effectively invalid (based on learning more about how etcd in proxy mode functions since the ticket was filed).

Fixing the authentication problem is just uncommenting those two variables and ensuring etcd's TLS has been correctly configured, as you have already added the client side parts of mutual TLS to K8S and flannel in this PR!

mjura replied:
Ok, I've added new commit to this PR https://github.com/kubic-project/salt/pull/102/commits/197d164e3318d28077e4f34267aee19a129556bc

mjura commented:
This change require update for e2e gating tests, https://github.com/kubic-project/e2e-tests/pull/61

MaximilianMeister commented:
`e2e-tests` are green in combination with https://github.com/kubic-project/e2e-tests/pull/61

mjura commented:
@kiall, @PI-Victor, @ereslibre, @flavio: any comments from you guys? can we merge it?

kiall commented:
Sure - I'll test this combined with the e2e-tests PR now!

PI-Victor commented:
i have nothing to add to this PR, can we merge it so we can close out the P1?

kiall commented:
Between VPN issues killing the ISO download on me over + over, and what I assume are some other issues on my workstation, I've not been able to get this to actually start up. I don't believe the issues I'm hitting are related to this change though, something, somewhere, is busted on my workstation.
